qiita_pet/templates/sitebase.html

```diff
@@ -93,45 +99,22 @@
       }
 
       function overlay_check() {
-        // from http://stackoverflow.com/q/5916900/379593
-        var ua=navigator.userAgent,tem,M=ua.match(/(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i) || [];
-        if(/trident/i.test(M[1])){
-          tem=/\brv[ :]+(\d+)/g.exec(ua) || [];
-          return 'IE '+(tem[1]||'');
-        }
-        if(M[1]==='Chrome'){
-          tem=ua.match(/\bOPR\/(\d+)/)
-          if(tem!=null)   {return 'Opera '+tem[1];}
-        }
-        M=M[2]? [M[1], M[2]]: [navigator.appName, navigator.appVersion, '-?'];
-        if((tem=ua.match(/version\/(\d+)/i))!=null) {M.splice(1,1,tem[1]);}
-        objbrowserName = M[0];
-        objBrMajorVersion = M[1];
-
-        if ((objbrowserName == "Chrome"  && objBrMajorVersion < {{allowed_min_browser_versions['chrome']}})
-            {% if 'firefox' in allowed_min_browser_versions %}
-              || (objbrowserName == "Firefox" && objBrMajorVersion < {{allowed_min_browser_versions['firefox']}})
-            {% else %}
-              || (objbrowserName == "Firefox")
-            {% end %}
-            {% if 'safari' in allowed_min_browser_versions %}
-              || (objbrowserName == "Safari"  && objBrMajorVersion < {{allowed_min_browser_versions['safari']}})
-            {% else %}
-              || (objbrowserName == "Safari")
-            {% end %}
-            {% if 'IE' in allowed_min_browser_versions %}
-              || ((objbrowserName == "IE" || objbrowserName == "MSIE") && objBrMajorVersion < {{allowed_min_browser_versions['IE']}})
-            {% else %}
-              || (objbrowserName == "IE" || objbrowserName == "MSIE")
-            {% end %}
-          ) {
-          // Show overlay because too old a browser
-          ol = document.getElementById("overlay");
-          ol.style.visibility = "visible";
-          navbar = document.getElementById("navigation-bar");
-          navbar.style.visibility = "hidden";
-          jumbotron = document.getElementById("jumbotron");
-          jumbotron.style.visibility = "hidden";
+        var browser_name = platform.name;
+        var browser_major_version = parseInt(platform.version.split("."));
+        if (!(browser_name in allowed_min_browser_versions) ||
+            browser_major_version < allowed_min_browser_versions[browser_name]['version']) {
+          var bowser_text = '';
+          // addding valid browsers
+          for (k in allowed_min_browser_versions) {
+            bowser_text += '<p id="explanation"><strong>' + k + ': ' +
+              allowed_min_browser_versions[k]['version'] + '.</strong> <a target="_blank" href="' +
+              allowed_min_browser_versions[k]['download'] + '">Download page</a></p>';
+          }
+          $('#overlay').append(bowser_text);
```

Comment:
might be a reason you're naming this bowser_text, instead of browser_text. If so, it might be good to put it in a comment; it looks like a spelling error.

Reply:
typo + copy/paste ... changing.